src/ClimaAtmos/Dycore/src/VanillaAtmosDiscretizations_kernels.jl

```diff
@@ -846,17 +924,45 @@ function facerhs!(::Val{dim}, ::Val{N}, ::Val{nmoist}, ::Val{ntrace},
 
         ρMinv = 1 / ρM
         uM, vM, wM = ρMinv * UM, ρMinv * VM, ρMinv * WM
-
         bc = elemtobndy[f, e]
-        PM = gdm1*(EM - (UM^2 + VM^2 + WM^2)/(2*ρM) - ρM*gravity*yorzM)
+        
+        for m = 1:nmoist
+            s = _nstate + m 
+            q_mM[m] = Q[vidM, s, eM] / ρM
+        end
+
+        e_intM = EM * ρMinv - ((UM^2 + VM^2+ WM^2)/(2*ρM) + ρM * gravity * yorzM) / ρM
+
+        TM = MoistThermodynamics.saturation_adjustment(e_intM, ρM, q_mM[1])
+        q_lM, q_iM = MoistThermodynamics.phase_partitioning_eq(TM, ρM, q_mM[1])
+
+        R_gasM, ~, ~, γM = MoistThermodynamics.moist_gas_constants(q_mM[1], q_lM, q_iM)
```

Comment:
use `_` instead of `~` for unnecessary arguments (this is reserved in Julia for this purpose).